src/mpid/ch4/shm/src/shm_init.c

```diff
@@ -38,8 +38,10 @@ int MPIDI_SHM_mpi_init_hook(int rank, int size, int *tag_bits)
     ret = MPIDI_POSIX_mpi_init_hook(rank, size, tag_bits);
     MPIR_ERR_CHECK(ret);
 
-    ret = MPIDI_IPC_mpi_init_hook(rank, size, tag_bits);
-    MPIR_ERR_CHECK(ret);
+    if (MPIR_CVAR_ENABLE_GPU) {
+        ret = MPIDI_IPC_mpi_init_hook(rank, size, tag_bits);
```

Comment:
This will disable XPMEM if there is no GPU, which I don't think should happen.

Reply:
Good point. I should move it to just where the gpu gets initialized.

Reply:
Fixed. Now it is around `MPIDI_GPU_mpi_init_hook`.